File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/ExternalVendorId.java
package com.attentive.androidsdk;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonValue;

@JsonInclude(JsonInclude.Include.NON_NULL)
class ExternalVendorId {
    private Vendor vendor;
    private String id;
    private String name;

    public Vendor getVendor() {
        return vendor;
    }

    public void setVendor(Vendor vendor) {
        this.vendor = vendor;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public enum Vendor {
        SHOPIFY("0"),
        KLAVIYO("1"),
        CLIENT_USER("2"),
        CUSTOM_USER("6");

        private final String id;

        Vendor(String id) {
            this.id = id;
        }

        @JsonValue
        String getVendorId() {
            return id;
        }
    }
}

File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/ExternalVendorId.kt
package com.attentive.androidsdk

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.descriptors.PrimitiveKind
import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
import kotlinx.serialization.descriptors.SerialDescriptor
import kotlinx.serialization.encoding.Decoder
import kotlinx.serialization.encoding.Encoder

// Custom serializer for Vendor enum
object VendorSerializer : KSerializer<ExternalVendorId.Vendor> {
    override val descriptor: SerialDescriptor =
        PrimitiveSerialDescriptor("Vendor", PrimitiveKind.STRING)

    override fun serialize(
        encoder: Encoder,
        value: ExternalVendorId.Vendor,
    ) {
        encoder.encodeString(value.vendorId)
    }

    override fun deserialize(decoder: Decoder): ExternalVendorId.Vendor {
        val vendorId = decoder.decodeString()
        return ExternalVendorId.Vendor.values().firstOrNull { it.vendorId == vendorId }
            ?: throw IllegalArgumentException("Unknown Vendor ID: $vendorId")
    }
}

@OptIn(ExperimentalSerializationApi::class)
@Serializable
open class ExternalVendorId {
    var vendor: Vendor? = null
    var id: String? = null
    var name: String? = null

    @Serializable(with = VendorSerializer::class)
    enum class Vendor(val vendorId: String) {
        SHOPIFY("0"),
        KLAVIYO("1"),
        CLIENT_USER("2"),
        CUSTOM_USER("6"),
    }
}
